src/rgw/rgw_aws.cc

```diff
@@ -0,0 +1,315 @@
+#include "rgw_url.h"
+#include "rgw_aws.h"
+#include "rgw_arn.h"
+#include <string>
+#include <iostream>
+#include <utility>
+#include "common/dout.h"
+#include <aws/core/Aws.h>
+#include <aws/sns/SNSClient.h>
+#include <aws/lambda/LambdaClient.h>
+#include <aws/lambda/model/InvokeRequest.h>
+#include <boost/lockfree/queue.hpp>
+#include <aws/core/utils/ARN.h>
+#include <aws/core/auth/AWSCredentials.h>
+#include <aws/sns/model/PublishRequest.h>
+
+#define dout_subsys ceph_subsys_rgw
+
+namespace rgw::aws {
+  static const int STATUS_OK = 1;
+  static const int STATUS_FULL = 2;
+  static const int STATUS_MANAGER_STOPPED = 3;
+  static const int STATUS_CLIENT_CLOSED = 4;
+
+  struct client_id_t {
+    const std::string accessKey;
```

Comment:
please refer https://github.com/ceph/ceph/blob/master/CodingStyle#L27 for our naming conventions.